form/src/org/netbeans/modules/form/layoutsupport/griddesigner/GridDesigner.java

```diff
@@ -264,7 +264,8 @@ private void setupUndoRedoShortcut(boolean undo, Action ourAction) {
         KeyStroke ks = null;
         FileObject undoRedoFO = FileUtil.getConfigFile(undo ? "Menu/Edit/org-openide-actions-UndoAction.shadow" : "Menu/Edit/org-openide-actions-RedoAction.shadow"); // NOI18N
         if (undoRedoFO != null) {
-            ContextAwareAction undoRedoAction = SystemAction.get(undo ? org.openide.actions.UndoAction.class : org.openide.actions.RedoAction.class);
+            Class undoClass = undo ? org.openide.actions.UndoAction.class : org.openide.actions.RedoAction.class;
```

Comment:
This will generate an unchecked warning. `Class<?>` would prevent that, I believe.

Reply:
@JaroslavTulach I tried both, both failed to build with cast and without cast. Same build error under with and without cast. Seems how I was doing it may generate warning but builds.

```Class<?>```

```java
src/org/netbeans/modules/form/layoutsupport/griddesigner/GridDesigner.java:268: error: method get in class SystemAction cannot be applied to given types;
            ContextAwareAction undoRedoAction = SystemAction.get(undoClass);
                                                            ^
  required: Class<T>
  found: Class<CAP#1>
  reason: inference variable T has incompatible bounds
    equality constraints: CAP#1
    upper bounds: SystemAction
  where T is a type-variable:
    T extends SystemAction declared in method <T>get(Class<T>)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends Object from capture of ?
```

```Class<? extends ContextAwareAction>```

```java
src/org/netbeans/modules/form/layoutsupport/griddesigner/GridDesigner.java:268: error: method get in class SystemAction cannot be applied to given types;
            ContextAwareAction undoRedoAction = SystemAction.get(undoClass);
                                                            ^
  required: Class<T>
  found: Class<CAP#1>
  reason: inference variable T has incompatible bounds
    equality constraints: CAP#1
    upper bounds: SystemAction
  where T is a type-variable:
    T extends SystemAction declared in method <T>get(Class<T>)
  where CAP#1 is a fresh type-variable:
    CAP#1 extends ContextAwareAction from capture of ? extends ContextAwareAction
```

I do not get an unchecked warning from GridDesigner.java. But the project has many unchecked warning, when adding in such compiler flag.

```java
 * Compiling ...
src/org/netbeans/modules/form/fakepeer/FakeComponentPeer.java:26: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:24: warning: ButtonPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ButtonPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:25: warning: CanvasPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CanvasPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:26: warning: CheckboxPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CheckboxPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:27: warning: ChoicePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ChoicePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:28: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:29: warning: ContainerPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ContainerPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:30: warning: LabelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.LabelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:31: warning: ListPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ListPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:32: warning: PanelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.PanelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:33: warning: ScrollPanePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollPanePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:34: warning: ScrollbarPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollbarPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:35: warning: TextAreaPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextAreaPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:36: warning: TextComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:37: warning: TextFieldPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextFieldPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeer.java:23: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakeComponentPeer.java:26: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:24: warning: ButtonPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ButtonPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:25: warning: CanvasPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CanvasPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:26: warning: CheckboxPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CheckboxPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:27: warning: ChoicePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ChoicePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:28: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:29: warning: ContainerPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ContainerPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:30: warning: LabelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.LabelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:31: warning: ListPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ListPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:32: warning: PanelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.PanelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:33: warning: ScrollPanePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollPanePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:34: warning: ScrollbarPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollbarPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:35: warning: TextAreaPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextAreaPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:36: warning: TextComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:37: warning: TextFieldPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextFieldPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeer.java:23: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakeComponentPeer.java:26: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:24: warning: ButtonPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ButtonPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:25: warning: CanvasPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CanvasPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:26: warning: CheckboxPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.CheckboxPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:27: warning: ChoicePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ChoicePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:28: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:29: warning: ContainerPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ContainerPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:30: warning: LabelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.LabelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:31: warning: ListPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ListPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:32: warning: PanelPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.PanelPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:33: warning: ScrollPanePeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollPanePeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:34: warning: ScrollbarPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ScrollbarPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:35: warning: TextAreaPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextAreaPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:36: warning: TextComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextComponentPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeerSupport.java:37: warning: TextFieldPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.TextFieldPeer;
                    ^
src/org/netbeans/modules/form/fakepeer/FakePeer.java:23: warning: ComponentPeer is internal proprietary API and may be removed in a future release
import java.awt.peer.ComponentPeer;
                    ^
src/org/netbeans/lib/awtextra/AbsoluteLayout.java:129: warning: [unchecked] unchecked call to put(K,V) as a member of the raw type Hashtable
        constraints.put(comp, constr);
                       ^
  where K,V are type-variables:
    K extends Object declared in class Hashtable
    V extends Object declared in class Hashtable
src/org/netbeans/modules/form/RADComponent.java:790: warning: [unchecked] unchecked cast
               (T)prop : null;
                  ^
  required: T
  found:    Property
  where T is a type-variable:
    T extends Object declared in method <T>getPropertyByName(String,Class<? extends T>,boolean)
src/org/netbeans/modules/form/FormProperty.java:123: warning: [unchecked] unchecked call to Property(Class<T>) as a member of the raw type Property
        super(type);
             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormProperty.java:134: warning: [unchecked] unchecked call to Property(Class<T>) as a member of the raw type Property
        super(type);
             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormProperty.java:146: warning: [unchecked] unchecked call to Property(Class<T>) as a member of the raw type Property
        super(type);
             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormProperty.java:159: warning: [unchecked] unchecked call to Property(Class<T>) as a member of the raw type Property
        super(type);
             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/RADProperty.java:117: warning: ComponentPeer is internal proprietary API and may be removed in a future release
        java.awt.peer.ComponentPeer scrollbarPeerHack =
                     ^
src/org/netbeans/modules/form/RADProperty.java:248: warning: [unchecked] unchecked call to getMethod(String,Class<?>...) as a member of the raw type Class
            Method method = enumClass.getMethod("values"); // NOI18N
                                               ^
src/org/netbeans/modules/form/ComponentChooserEditor.java:335: warning: [unchecked] unchecked call to isAssignableFrom(Class<?>) as a member of the raw type Class
            match = beanTypes[i].isAssignableFrom(comp.getBeanClass());
                                                 ^
src/org/netbeans/modules/form/RADVisualComponent.java:319: warning: [unchecked] unchecked conversion
                l = new ArrayList(constraintsProperties.length+1);
                    ^
  required: List<Property>
  found:    ArrayList
src/org/netbeans/modules/form/RADVisualComponent.java:322: warning: [unchecked] unchecked conversion
                l = new ArrayList(1);
                    ^
  required: List<Property>
  found:    ArrayList
src/org/netbeans/modules/form/RADVisualComponent.java:444: warning: [unchecked] unchecked call to ReadWrite(String,Class<T>,String,String) as a member of the raw type ReadWrite
            super(name, type, displayName, shortDescription);
                 ^
  where T is a type-variable:
    T extends Object declared in class ReadWrite
src/org/netbeans/modules/form/RADVisualComponent.java:458: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                setValue(getDefaultValue());
                        ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/ParametersPicker.java:89: warning: [unchecked] unchecked call to addItem(E) as a member of the raw type JComboBox
            beanCombo.addItem(FormUtils.getBundleString("CTL_CW_SelectBean")); // NOI18N
                             ^
  where E is a type-variable:
    E extends Object declared in class JComboBox
src/org/netbeans/modules/form/ParametersPicker.java:93: warning: [unchecked] unchecked call to addItem(E) as a member of the raw type JComboBox
                    beanCombo.addItem(
                                     ^
  where E is a type-variable:
    E extends Object declared in class JComboBox
src/org/netbeans/modules/form/ParametersPicker.java:96: warning: [unchecked] unchecked call to addItem(E) as a member of the raw type JComboBox
                    beanCombo.addItem(radComp.getName());
                                     ^
  where E is a type-variable:
    E extends Object declared in class JComboBox
src/org/netbeans/modules/form/PropertyPicker.java:62: warning: [unchecked] unchecked call to addItem(E) as a member of the raw type JComboBox
                componentsCombo.addItem(
                                       ^
  where E is a type-variable:
    E extends Object declared in class JComboBox
src/org/netbeans/modules/form/PropertyPicker.java:65: warning: [unchecked] unchecked call to addItem(E) as a member of the raw type JComboBox
                componentsCombo.addItem(radComp.getName());
                                       ^
  where E is a type-variable:
    E extends Object declared in class JComboBox
src/org/netbeans/modules/form/PropertyPicker.java:125: warning: [unchecked] unchecked call to setListData(E[]) as a member of the raw type JList
            propertyList.setListData(new Object [0]);
                                    ^
  where E is a type-variable:
    E extends Object declared in class JList
src/org/netbeans/modules/form/PropertyPicker.java:165: warning: [unchecked] unchecked call to setListData(E[]) as a member of the raw type JList
            propertyList.setListData(listItems);
                                    ^
  where E is a type-variable:
    E extends Object declared in class JList
src/org/netbeans/modules/form/EventProperty.java:49: warning: [unchecked] unchecked call to ReadWrite(String,Class<T>,String,String) as a member of the raw type ReadWrite
        super(eventId,
             ^
  where T is a type-variable:
    T extends Object declared in class ReadWrite
src/org/netbeans/modules/form/EventProperty.java:179: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                                        prop.setValue(newName);
                                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/codestructure/CodeStructure.java:194: warning: [unchecked] unchecked call to add(E) as a member of the raw type List
            list.add(it.next());
                    ^
  where E is a type-variable:
    E extends Object declared in interface List
src/org/netbeans/modules/form/codestructure/CodeStructure.java:353: warning: [unchecked] unchecked call to add(E) as a member of the raw type List
            usingObjects.add(it.next());
                            ^
  where E is a type-variable:
    E extends Object declared in interface List
src/org/netbeans/modules/form/FormJavaSource.java:107: warning: [unchecked] unchecked call to set(T) as a member of the raw type ThreadLocal
                tl.set(javaContext);
                      ^
  where T is a type-variable:
    T extends Object declared in class ThreadLocal
src/org/netbeans/modules/form/FormUtils.java:545: warning: [unchecked] unchecked call to addElement(E) as a member of the raw type DefaultListModel
                newListModel.addElement(cloneObject(listModel.getElementAt(i), formModel));
                                       ^
  where E is a type-variable:
    E extends Object declared in class DefaultListModel
src/org/netbeans/modules/form/FormUtils.java:554: warning: [unchecked] unchecked call to addElement(E) as a member of the raw type DefaultComboBoxModel
                newComboModel.addElement(cloneObject(comboModel.getElementAt(i), formModel));
                                        ^
  where E is a type-variable:
    E extends Object declared in class DefaultComboBoxModel
src/org/netbeans/modules/form/FormUtils.java:886: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                else tnProp.setValue(copiedValue);
                                    ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormUtils.java:999: warning: [unchecked] unchecked call to getMethod(String,Class<?>...) as a member of the raw type Class
                method = targetClass.getMethod(method.getName(),
                                              ^
src/org/netbeans/modules/form/FormModelEvent.java:559: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                            prop.setValue(getOldPropertyValue());
                                         ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:581: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                            prop.setValue(getNewPropertyValue());
                                         ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:601: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(getOldPropertyValue());
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:616: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(getNewPropertyValue());
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:711: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                    prop.setValue(getOldPropertyValue());
                                 ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:723: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                    prop.setValue(getNewPropertyValue());
                                 ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:801: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                            props[i].setValue(getOldPropertyValue());
                                             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:845: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                            props[i].setValue(getNewPropertyValue());
                                             ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:876: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(null);
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:899: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                    prop.setValue(getEventHandler());
                                 ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:922: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                    prop.setValue(getEventHandler());
                                 ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:946: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(null);
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:967: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(getOldEventHandler());
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/FormModelEvent.java:989: warning: [unchecked] unchecked call to setValue(T) as a member of the raw type Property
                        prop.setValue(getNewEventHandler());
                                     ^
  where T is a type-variable:
    T extends Object declared in class Property
src/org/netbeans/modules/form/MetaComponentCreator.java:1630: warning: [unchecked] unchecked call to DefaultComboBoxModel(E[]) as a member of the raw type DefaultComboBoxModel
                Object propValue = new DefaultComboBoxModel(new String[] {
                                   ^
  where E is a type-variable:
    E extends Object declared in class DefaultComboBoxModel
src/org/netbeans/modules/form/MetaComponentCreator.java:1644: warning: [unchecked] unchecked call to addElement(E) as a member of the raw type DefaultListModel
                    defaultModel.addElement(prefix + i); // NOI18N
                                           ^
  where E is a type-variable:
    E extends Object declared in class DefaultListModel
src/org/netbeans/modules/form/RADVisualContainer.java:172: warning: [unchecked] unchecked call to isAssignableFrom(Class<?>) as a member of the raw type Class
                   || (layoutClass2.isAssignableFrom(layoutClass1)
                                                    ^
src/org/netbeans/modules/form/RADVisualContainer.java:174: warning: [unchecked] unchecked call to isAssignableFrom(Class<?>) as a member of the raw type Class
                   || (layoutClass1.isAssignableFrom(layoutClass2)
                                                    ^
src/org/netbeans/modules/form/layoutsupport/LayoutNode.java:171: warning: [unchecked] unchecked method invocation: method doPrivileged in class AccessController is applied to given types
        AccessController.doPrivileged(new PrivilegedAction() {
                                     ^
  required: PrivilegedAction<T>
  found: <anonymous PrivilegedAction>
  where T is a type-variable:
    T extends Object declared in method <T>doPrivileged(PrivilegedAction<T>)
src/org/netbeans/modules/form/layoutsupport/LayoutNode.java:171: warning: [unchecked] unchecked conversion
        AccessController.doPrivileged(new PrivilegedAction() {
                                      ^
  required: PrivilegedAction<T>
  found:    <anonymous PrivilegedAction>
  where T is a type-variable:
    T extends Object declared in method <T>doPrivileged(PrivilegedAction<T>)
src/org/netbeans/modules/form/layoutdesign/LayoutComponent.java:220: warning: [unchecked] unchecked conversion
               Collections.unmodifiableList(subComponents) : Collections.EMPTY_LIST;
                                                                        ^
  required: List<LayoutComponent>
  found:    List
src/org/netbeans/modules/form/palette/PaletteItem.java:118: warning: [unchecked] unchecked method invocation: constructor <init> in class ClassSource is applied to given types
        setComponentClassSource(new ClassSource((className == null) ? null : className.trim(), Collections.EMPTY_LIST, typeParameters));
                                ^
  required: String,Collection<? extends Entry>,String
  found: String,List,String
src/org/netbeans/modules/form/palette/PaletteItem.java:118: warning: [unchecked] unchecked conversion
        setComponentClassSource(new ClassSource((className == null) ? null : className.trim(), Collections.EMPTY_LIST, typeParameters));
                                                                                                          ^
  required: Collection<? extends Entry>
  found:    List
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: Some input files additionally use unchecked or unsafe operations.
100 warnings
>>> Source compiled.

```

Reply:
The JDK9 change the type resolution logic, the following works:
```java
            Class<? extends SystemAction> undoClass = undo ? org.openide.actions.UndoAction.class : org.openide.actions.RedoAction.class;
            ContextAwareAction undoRedoAction = (ContextAwareAction) SystemAction.get(undoClass);
```

Reply:
Ok thanks that does work! I updated and force pushed the changes.